addons/web/controllers/binary.py

```diff
@@ -247,46 +255,47 @@ def company_logo(self, dbname=None, **kw):
-                        image_data = io.BytesIO(image_base64)
-                        mimetype = guess_mimetype(image_base64, default='image/png')
-                        imgext = '.' + mimetype.split('/')[1]
-                        if imgext == '.svg+xml':
-                            imgext = '.svg'
-                        response = send_file(
-                            image_data,
-                            request.httprequest.environ,
-                            download_name=imgname + imgext,
-                            mimetype=mimetype,
-                            last_modified=row[1],
-                            response_class=Response,
-                        )
-                    else:
-                        response = http.Stream.from_path(file_path('web/static/img/nologo.png')).get_response()
-            except Exception:
+                company = int(kw['company']) if kw and kw.get('company') else False
+                if company:
+                    request.env.cr.execute("""SELECT logo_web, write_date
+                                    FROM res_company
+                                    WHERE id = %s
+                                """, (company,))
+                else:
+                    request.env.cr.execute("""SELECT c.logo_web, c.write_date
+                                    FROM res_users u
+                                LEFT JOIN res_company c
+                                        ON c.id = u.company_id
+                                    WHERE u.id = %s
+                                """, (uid,))
+                row = request.env.cr.fetchone()
+                if row and row[0]:
+                    image_base64 = base64.b64decode(row[0])
+                    image_data = io.BytesIO(image_base64)
+                    mimetype = guess_mimetype(image_base64, default='image/png')
+                    imgext = '.' + mimetype.split('/')[1]
+                    if imgext == '.svg+xml':
+                        imgext = '.svg'
+                    response = send_file(
+                        image_data,
+                        request.httprequest.environ,
+                        download_name=imgname + imgext,
+                        mimetype=mimetype,
+                        last_modified=row[1],
+                        response_class=Response,
+                    )
+                else:
+                    response = http.Stream.from_path(file_path('web/static/img/nologo.png')).get_response()
+            except Exception as e:
+                _logger.warning(e)
                 response = http.Stream.from_path(file_path(f'web/static/img/{imgname}{imgext}')).get_response()
 
         return response
 
-    @http.route(['/web/sign/get_fonts', '/web/sign/get_fonts/<string:fontname>'], type='json', auth='public')
+    @http.route([
+        '/web/sign/get_fonts',
+        '/web/sign/get_fonts/<string:fontname>',
+    ], type='json', auth='none')
```

Comment:
Looks like this one can be made no-db
```suggestion
    ], type='json', no_db=True)
```